src/main/java/uk/gov/hmcts/reform/hmc/service/PartiesNotifiedServiceImpl.java

```diff
@@ -37,12 +37,14 @@ public PartiesNotifiedServiceImpl(HearingResponseRepository hearingResponseRepos
     }
 
     @Override
-    public void getPartiesNotified(Long hearingId, int responseVersion, PartiesNotified partiesNotified) {
+    public void getPartiesNotified(Long hearingId, Integer requestVersion,
+                                   LocalDateTime receivedDateTime, PartiesNotified partiesNotified) {
         hearingIdValidator.validateHearingId(hearingId, PARTIES_NOTIFIED_ID_NOT_FOUND);
-        HearingResponseEntity hearingResponseEntity = hearingResponseRepository.getHearingResponse(hearingId);
-        if (hearingResponseEntity.getResponseVersion() != responseVersion) {
-            throw new PartiesNotifiedNotFoundException(PARTIES_NOTIFIED_RESPONSE_VERSION_MISMATCH);
-        } else if (hearingResponseEntity.getResponseVersion().equals(responseVersion)
+        HearingResponseEntity hearingResponseEntity =
+                hearingResponseRepository.getHearingResponse(hearingId, requestVersion, receivedDateTime);
+        if (null == hearingResponseEntity) {
+            throw new PartiesNotifiedNotFoundException(PARTIES_NOTIFIED_NO_SUCH_RESPONSE);
+        } else if (hearingResponseEntity.getRequestVersion().equals(requestVersion)
             && hearingResponseEntity.getPartiesNotifiedDateTime() != null) {
```

Comment:
```suggestion
        } else if (hearingResponseEntity.getPartiesNotifiedDateTime() != null) {
```
Not sure that `requestVersion` check is giving us anything? The search criteria includes `requestVersion` so if it's not `null` then it'll have to be correct.